lib/phoenix/transports/websocket.ex

```diff
@@ -50,29 +50,17 @@ defmodule Phoenix.Transports.WebSocket do
 
   import Plug.Conn, only: [fetch_query_params: 1, send_resp: 3]
 
-  alias Phoenix.Socket.Broadcast
-  alias Phoenix.Socket.Transport
-
   @doc false
-  def init(%Plug.Conn{method: "GET"} = conn, {endpoint, handler, transport}) do
-    {_, opts} = handler.__transport__(transport)
-
-    conn =
-      conn
-      |> code_reload(opts, endpoint)
-      |> Plug.Conn.fetch_query_params
-      |> Transport.transport_log(opts[:transport_log])
-      |> Transport.force_ssl(handler, endpoint, opts)
-      |> Transport.check_origin(handler, endpoint, opts)
-
-    case conn do
-      %{halted: false} = conn ->
-        params     = conn.params
-        serializer = Keyword.fetch!(opts, :serializer)
+  def init(%Plug.Conn{method: "GET"} = conn, {endpoint, socket_handler, transport}) do
+    {_, opts} = socket_handler.__transport__(transport)
 
-        case Transport.connect(endpoint, handler, transport, __MODULE__, serializer, params) do
-          {:ok, socket} ->
-            {:ok, conn, {__MODULE__, {socket, opts}}}
+    conn
+    |> Phoenix.Transports.Utils.init_plug_conn(endpoint, socket_handler, transport)
+    |> case do
```

Comment:
Ah, nicely spotted! I'll try to play and see how it looks with `with`, it's a good chance to try it under my fingers :-)